impl/src/main/java/org/ehcache/internal/persistence/DefaultLocalPersistenceService.java

```diff
@@ -262,7 +263,27 @@ private static boolean recursiveDelete(File file) {
     }
     return true;
   }
+  
+  private static boolean tryRecursiveDelete(File file) {
+    boolean success = true;
+    for (int i = 0; i < 5; i++) {
+      success = recursiveDelete(file);
+      if (success) {
+        break;
+      } else {
+        System.gc();
+        System.runFinalization();
 
+        try {
+          Thread.sleep(50);
+        } catch (InterruptedException e) {
+          // do nothing ?
```

Comment:
Probably best to store the interrupted status in a local boolean here and reassert on method exit.